<a href="https://colab.research.google.com/github/shrivatsashetty/Projects/blob/main/Synamedia_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Import modules**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
# %matplotlib notebook

In [2]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [3]:
main_df = pd.read_csv("/content/drive/MyDrive/Data-Science/2DaysViewingSession.csv")

# **Tabular form**

In [4]:
main_df

state                          event_start_tz  \
0             Selangor  2023-08-27 11:03:13.686 Asia/Singapore   
1         Kuala Lumpur  2023-08-27 10:48:56.982 Asia/Singapore   
2                Johor  2023-08-27 10:56:51.851 Asia/Singapore   
3         Kuala Lumpur  2023-08-27 10:31:40.717 Asia/Singapore   
4               Penang  2023-08-27 10:24:16.194 Asia/Singapore   
...                ...                                     ...   
26011962  Kuala Lumpur  2023-08-28 15:04:59.497 Asia/Singapore   
26011963  Kuala Lumpur  2023-08-28 15:05:03.023 Asia/Singapore   
26011964  Kuala Lumpur  2023-08-28 15:05:16.592 Asia/Singapore   
26011965       Sarawak  2023-08-28 14:59:50.687 Asia/Singapore   
26011966  Kuala Lumpur  2023-08-28 15:00:12.829 Asia/Singapore   

          duration_seconds content_type                      content  \
0                       40          VOD                  Ombak Rindu   
1                      424          VOD    Pada Zaman Dahulu S4 Ep08   
2                      398          VOD         Upin & Ipin S16 Ep34   
3                     1411          VOD         Tanah Kubur S13 Ep08   
4                     2526          VOD        Narcotics Heroes Ep17   
...                    ...          ...                          ...   
26011962                 1         CDVR          Ghost of Relativity   
26011963               906         CDVR          Ghost of Relativity   
26011964               362          VOD         Upin & Ipin S15 Ep34   
26011965              1241          VOD               Mechamato Ep01   
26011966               672          VOD  Tayo The Little Bus S5 Ep13   

          content_source  
0               astro.2d  
1               astro.7d  
2              astro.30d  
3              astro.30d  
4              astro.30d  
...                  ...  
26011962  TVB CLASSIC HD  
26011963  TVB CLASSIC HD  
26011964       astro.30d  
26011965       astro.30d  
26011966        astro.7d  

[26011967 rows x 6 columns]

# *main dataframe shape*

In [5]:
main_df.shape

(26011967, 6)

### ***the main dataframe contains 2,60,11,967 records***

# main dataset columns

In [6]:
i = 1
print("The dataset contains the following columns")
for column in main_df.columns:
  print(i,".",column)
  i+=1

The dataset contains the following columns
1 . state
2 . event_start_tz
3 . duration_seconds
4 . content_type
5 . content
6 . content_source


In [7]:
main_df.describe()

duration_seconds
count      2.601197e+07
mean       7.747659e+02
std        1.274039e+03
min        0.000000e+00
25%        5.700000e+01
50%        2.490000e+02
75%        9.730000e+02
max        3.239500e+04

In [8]:
main_df.isna().sum()

state               13493121
event_start_tz             0
duration_seconds           0
content_type               0
content               110136
content_source          3780
dtype: int64

In [9]:
cleaned_df = main_df.dropna()

In [10]:
cleaned_df.shape

(12466618, 6)

In [11]:
type(cleaned_df.shape)

tuple

# difference b/w new and orignal data frame

In [12]:
records_differenece = main_df.shape[0] - cleaned_df.shape[0]

In [13]:
percent_record_difference = (records_differenece / main_df.shape[0]) * 100

### percentage difference in records after dropping rows

In [14]:
percent_record_difference

52.073528311027

In [15]:
cleaned_df.describe()

duration_seconds
count      1.246662e+07
mean       5.246197e+02
std        1.160204e+03
min        0.000000e+00
25%        1.000000e+01
50%        1.250000e+02
75%        4.280000e+02
max        3.239500e+04

In [16]:
cleaned_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12466618 entries, 0 to 26011966
Data columns (total 6 columns):
 #   Column            Dtype 
---  ------            ----- 
 0   state             object
 1   event_start_tz    object
 2   duration_seconds  int64 
 3   content_type      object
 4   content           object
 5   content_source    object
dtypes: int64(1), object(5)
memory usage: 665.8+ MB


In [17]:
uniq_states = cleaned_df['state'].value_counts()

### List of distinct states

In [18]:
uniq_states


Kuala Lumpur                  6878719
Selangor                      2634243
Penang                         751319
Johor                          722522
Perak                          404518
Sarawak                        228747
Negeri Sembilan                219438
Sabah                          208386
Melaka                         119456
Kedah                           94178
Kelantan                        62062
Pahang                          52238
UNKNOWN                         48884
Terengganu                      31681
Putrajaya                        7271
Perlis                           1869
Karnataka                         379
Labuan                            370
ERROR                             169
Maharashtra                       129
Seoul                              10
Uttar Pradesh                       6
Occitanie                           5
England                             5
Victoria                            3
Jerusalem                           3
Tamil Nadu  

# List of actual Malaysian states plus federal territories

In [19]:
states_list = ["Perlis", "Kedah", "Penang", "Perak", "Selangor", "Negeri Sembilan",
               "Melaka", "Johor", "Kelantan", "Terengganu", "Pahang", "Kuala Lumpur", "Putrajaya", "Sabah", "Sarawak", "Labuan" ]

In [20]:
len(states_list)

16

# filtering duplicate state if any

In [21]:
uniq_states_list = list(set(states_list))

In [22]:
for state in uniq_states_list:
  print(state)

Selangor
Labuan
Pahang
Sarawak
Kelantan
Kuala Lumpur
Johor
Perak
Kedah
Sabah
Perlis
Terengganu
Melaka
Penang
Putrajaya
Negeri Sembilan


In [23]:
len(uniq_states_list)

16

In [24]:
print("Total distinct states:",uniq_states.count())

Total distinct states: 31


In [25]:
org_distinct_states = main_df['state'].value_counts()

In [26]:
org_distinct_states.count()

31

# Filtering  non Malaysian states

In [27]:
cleaned_df = cleaned_df[cleaned_df['state'].isin(uniq_states_list)]

In [28]:
type(cleaned_df['state'].isin(uniq_states_list))

pandas.core.series.Series

In [29]:
cleaned_df_states = cleaned_df['state'].value_counts()

In [30]:
cleaned_df_states

Kuala Lumpur       6878719
Selangor           2634243
Penang              751319
Johor               722522
Perak               404518
Sarawak             228747
Negeri Sembilan     219438
Sabah               208386
Melaka              119456
Kedah                94178
Kelantan             62062
Pahang               52238
Terengganu           31681
Putrajaya             7271
Perlis                1869
Labuan                 370
Name: state, dtype: int64

In [31]:
type(cleaned_df_states)

pandas.core.series.Series

In [32]:
cleaned_df_states.sum()

12417017

In [33]:
cleaned_df_states.count()

16

### hence the **final version of cleaned_df** contains list of only 16 states which is actual no of states in malaysia

In [34]:
cleaned_df.shape

(12417017, 6)

## Till here we have a final version of cleaned dataframe

# Now scaling down the dataset

In [35]:
sample_size = int(cleaned_df.shape[0] * 0.4)

In [36]:
sample_size

4966806

In [37]:
sampled_df = cleaned_df.sample(n=sample_size, random_state=42)

In [38]:
sampled_states_series = sampled_df["state"].value_counts()

In [39]:
sampled_states_series

Kuala Lumpur       2752122
Selangor           1052497
Penang              301065
Johor               289287
Perak               161315
Sarawak              91277
Negeri Sembilan      87627
Sabah                83336
Melaka               47890
Kedah                37831
Kelantan             24785
Pahang               21128
Terengganu           12850
Putrajaya             2883
Perlis                 755
Labuan                 158
Name: state, dtype: int64

In [40]:
cumilative_frequency_of_states = sampled_states_series.sum()

In [41]:
cumilative_frequency_of_states

4966806

In [42]:
statewise_frequency_percentages = {}
for state in sampled_states_series.index:
  statewise_frequency_percentages[state] = (sampled_states_series[state] / cumilative_frequency_of_states) * 100

In [43]:
statewise_frequency_percentages

{'Kuala Lumpur': 55.410297885602944,
 'Selangor': 21.190620289981126,
 'Penang': 6.061541360785986,
 'Johor': 5.824407073680752,
 'Perak': 3.247861905619023,
 'Sarawak': 1.8377403909071544,
 'Negeri Sembilan': 1.7642525196273018,
 'Sabah': 1.677858970130905,
 'Melaka': 0.964201138518396,
 'Kedah': 0.7616766187364676,
 'Kelantan': 0.4990128464852463,
 'Pahang': 0.4253840395618431,
 'Terengganu': 0.25871757423181013,
 'Putrajaya': 0.058045351479401444,
 'Perlis': 0.015200915840079116,
 'Labuan': 0.003181118811566226}

# list of distinct content_type (content_category)


In [44]:
content_categories = sampled_df['content_type'].value_counts()

In [45]:
content_categories

VOD                2933893
LINEAR              884036
CDVR                693233
Live Pause          183495
Play From Start     160127
IP Switch           112022
Name: content_type, dtype: int64

# list of distinct content (T.V Shows)

In [46]:
content_series = sampled_df['content'].value_counts()

 * ***the value_counts() method will always return a pandas series with unique indexes so no need to worry about the uniqueness of index names***

In [47]:
content_series

(L) BWF World Championships: Finals     71166
(L) PL: Newcastle vs Liverpool          65977
(L) PL: Sheffield vs Manchester City    44857
The Flash [Fast Premiere]               41814
Tiktok Awards Malaysia                  32574
                                        ...  
Indira Ep184                                1
Little Food Hunter S1 Ep07                  1
My Little Monkey King S1 Ep14               1
Room 104 S1 Ep09                            1
Tong Tong's Wonderland S3 Ep10              1
Name: content, Length: 55068, dtype: int64

In [48]:
len(content_series)

55068

## There are 55068 distinct shows. Below is the top 10 most occuring TV shows


In [49]:
content_series.head(10)

(L) BWF World Championships: Finals       71166
(L) PL: Newcastle vs Liverpool            65977
(L) PL: Sheffield vs Manchester City      44857
The Flash [Fast Premiere]                 41814
Tiktok Awards Malaysia                    32574
Super Singer Junior                       28424
(L) Formula 1 Main Race 2023              24791
FIBA: Latvia vs Lebanon                   21701
Start Music                               19582
(L) World Athletics Championships 2023    19151
Name: content, dtype: int64

## sports shows are most watched

In [50]:
distinct_contents_dictionary = content_series.to_dict()

In [51]:
# since the dictionary is too big lets only see the first few elements
first_few_elements = 100
for key, value in distinct_contents_dictionary.items():
  print(key, ":", distinct_contents_dictionary[key])
  first_few_elements -= 1
  if first_few_elements == 0:
    break

(L) BWF World Championships: Finals : 71166
(L) PL: Newcastle vs Liverpool : 65977
(L) PL: Sheffield vs Manchester City : 44857
The Flash [Fast Premiere] : 41814
Tiktok Awards Malaysia : 32574
Super Singer Junior : 28424
(L) Formula 1 Main Race 2023 : 24791
FIBA: Latvia vs Lebanon : 21701
Start Music : 19582
(L) World Athletics Championships 2023 : 19151
Buletin Utama : 18575
Pathaan : 17764
Varalaru : 17450
Rajuk Karina : 17199
Sepahtu Reunion Live S8 Ep07 : 16478
2000 Prime Talk : 16138
Evening Edition : 15551
Saregamapa Little Champs : 15218
Narcotics Heroes : 13426
(L) Miss Hong Kong Pageant 2023 : 13378
Baagyaletchumi Illatharisiyin Kadhai : 12520
Pandian Stores : 12238
Super Nabil Ep01 : 12130
Thamizhum Saraswathiyum : 11763
Puss In Boots: The Last Wish : 11729
Running Man : 11658
MasterChef Australia : 11592
MW3: Newcastle United vs Liverpool : 11012
(L) MSL 2023: PRK lwn KLC : 10873
Siragadikka Aasai : 9977
Aku Yang Kau Sakiti : 9716
Ethirneechal : 9076
(L) PL: Burnley vs Aston

# Total count of distinct TV shows i.e contents

In [52]:
len(distinct_contents_dictionary)

55068

In [53]:
datetime_format = "%Y-%m-%d %H:%M:%S.%f %Z"

# Convert the column to datetime using the specified format
sampled_df['event_start_tz'] = pd.to_datetime(sampled_df['event_start_tz'], format=datetime_format)

# Check the data type of the 'event_start_tz' column
print(sampled_df['event_start_tz'].dtype)


datetime64[ns, Asia/Singapore]


In [55]:
sampled_df['date'] = sampled_df['event_start_tz'].dt.date
sampled_df['time'] = sampled_df['event_start_tz'].dt.time
sampled_df['timezone'] = sampled_df['event_start_tz'].dt.tz
sampled_df['end_time'] = sampled_df['event_start_tz'] + pd.to_timedelta(sampled_df['duration_seconds'], unit='s')
sampled_df['timestamp'] = sampled_df['event_start_tz'].apply(lambda x: x.timestamp())
# Assuming you have a DataFrame named 'sampled_df' with a 'date' column

# Convert the 'date' column to a datetime format
sampled_df['date'] = pd.to_datetime(sampled_df['date'])

# Add a new 'day' column to represent the day of the week
sampled_df['day'] = sampled_df['date'].dt.day_name()



In [56]:
sampled_df.shape

(4966806, 12)

In [57]:
sampled_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4966806 entries, 14622656 to 21151475
Data columns (total 12 columns):
 #   Column            Dtype                         
---  ------            -----                         
 0   state             object                        
 1   event_start_tz    datetime64[ns, Asia/Singapore]
 2   duration_seconds  int64                         
 3   content_type      object                        
 4   content           object                        
 5   content_source    object                        
 6   date              datetime64[ns]                
 7   time              object                        
 8   timezone          object                        
 9   end_time          datetime64[ns, Asia/Singapore]
 10  timestamp         float64                       
 11  day               object                        
dtypes: datetime64[ns, Asia/Singapore](2), datetime64[ns](1), float64(1), int64(1), object(7)
memory usage: 492.6+ MB


In [58]:
sampled_df["event_start_tz"][0]


Timestamp('2023-08-27 11:03:13.686000+0800', tz='Asia/Singapore')

In [59]:
sampled_df

state                   event_start_tz  duration_seconds  \
14622656      Selangor 2023-08-28 19:37:11.438000+08:00               119   
24953294  Kuala Lumpur 2023-08-29 00:54:11.028000+08:00               378   
12353742      Selangor 2023-08-28 00:55:21.046000+08:00              1949   
4993584       Selangor 2023-08-27 14:08:21.937000+08:00                87   
20949672         Johor 2023-08-28 13:51:03.669000+08:00               128   
...                ...                              ...               ...   
13690390  Kuala Lumpur 2023-08-28 01:48:37.945000+08:00              2216   
9081794       Selangor 2023-08-27 20:02:38.647000+08:00                 0   
19056004  Kuala Lumpur 2023-08-28 17:35:44.997000+08:00                49   
20244015      Selangor 2023-08-28 22:16:24.330000+08:00                66   
21151475      Selangor 2023-08-28 12:07:07.003000+08:00                 3   

         content_type                         content  content_source  \
14622656          VOD     Lagu Didi & Friends S3 Ep31       astro.30d   
24953294          VOD             Upin & Ipin S5 Ep06       astro.30d   
12353742       LINEAR  (L) PL: Newcastle vs Liverpool        ASSP3 HD   
4993584        LINEAR        A Pillow Case of Mystery  TVB CLASSIC HD   
20949672          VOD     Lagu Didi & Friends S2 Ep08       astro.30d   
...               ...                             ...             ...   
13690390          VOD      Jangan Menangis Cinta Ep18       astro.30d   
9081794           VOD       The Flash [Fast Premiere]    astro.com.my   
19056004       LINEAR  Arsenal TV : Arsenal Vs Fulham     beIN SPORTS   
20244015          VOD         Where's Chicky? S2 Ep14        astro.7d   
21151475   Live Pause                     Berita 7:57      Awesome TV   

               date             time        timezone  \
14622656 2023-08-28  19:37:11.438000  Asia/Singapore   
24953294 2023-08-29  00:54:11.028000  Asia/Singapore   
12353742 2023-08-28  00:55:21.046000  Asia/Singapore   
4993584  2023-08-27  14:08:21.937000  Asia/Singapore   
20949672 2023-08-28  13:51:03.669000  Asia/Singapore   
...             ...              ...             ...   
13690390 2023-08-28  01:48:37.945000  Asia/Singapore   
9081794  2023-08-27  20:02:38.647000  Asia/Singapore   
19056004 2023-08-28  17:35:44.997000  Asia/Singapore   
20244015 2023-08-28  22:16:24.330000  Asia/Singapore   
21151475 2023-08-28  12:07:07.003000  Asia/Singapore   

                                 end_time     timestamp      day  
14622656 2023-08-28 19:39:10.438000+08:00  1.693223e+09   Monday  
24953294 2023-08-29 01:00:29.028000+08:00  1.693242e+09  Tuesday  
12353742 2023-08-28 01:27:50.046000+08:00  1.693155e+09   Monday  
4993584  2023-08-27 14:09:48.937000+08:00  1.693117e+09   Sunday  
20949672 2023-08-28 13:53:11.669000+08:00  1.693202e+09   Monday  
...                                   ...           ...      ...  
13690390 2023-08-28 02:25:33.945000+08:00  1.693159e+09   Monday  
9081794  2023-08-27 20:02:38.647000+08:00  1.693138e+09   Sunday  
19056004 2023-08-28 17:36:33.997000+08:00  1.693215e+09   Monday  
20244015 2023-08-28 22:17:30.330000+08:00  1.693232e+09   Monday  
21151475 2023-08-28 12:07:10.003000+08:00  1.693196e+09   Monday  

[4966806 rows x 12 columns]

## duration_seconds is the only numerical field hence lets see description of that field

In [60]:
sampled_df.describe()

duration_seconds     timestamp
count      4.966806e+06  4.966806e+06
mean       5.243420e+02  1.693172e+09
std        1.160789e+03  4.664119e+04
min        0.000000e+00  1.693063e+09
25%        1.000000e+01  1.693131e+09
50%        1.250000e+02  1.693180e+09
75%        4.270000e+02  1.693216e+09
max        3.239500e+04  1.693267e+09

In [61]:
sampled_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4966806 entries, 14622656 to 21151475
Data columns (total 12 columns):
 #   Column            Dtype                         
---  ------            -----                         
 0   state             object                        
 1   event_start_tz    datetime64[ns, Asia/Singapore]
 2   duration_seconds  int64                         
 3   content_type      object                        
 4   content           object                        
 5   content_source    object                        
 6   date              datetime64[ns]                
 7   time              object                        
 8   timezone          object                        
 9   end_time          datetime64[ns, Asia/Singapore]
 10  timestamp         float64                       
 11  day               object                        
dtypes: datetime64[ns, Asia/Singapore](2), datetime64[ns](1), float64(1), int64(1), object(7)
memory usage: 621.6+ MB


### ***sampled dataframe  is what % of orignal dataframe ?***

In [62]:
(sampled_df.shape[0] / main_df.shape[0]) * 100

19.0943114759449

### Now we consider only those shows whose watch duration is non-zero

In [63]:
nz_duartion_df = sampled_df[sampled_df["duration_seconds"] !=0]

In [64]:
nz_duartion_df

state                   event_start_tz  duration_seconds  \
14622656      Selangor 2023-08-28 19:37:11.438000+08:00               119   
24953294  Kuala Lumpur 2023-08-29 00:54:11.028000+08:00               378   
12353742      Selangor 2023-08-28 00:55:21.046000+08:00              1949   
4993584       Selangor 2023-08-27 14:08:21.937000+08:00                87   
20949672         Johor 2023-08-28 13:51:03.669000+08:00               128   
...                ...                              ...               ...   
21510289      Selangor 2023-08-28 21:41:16.559000+08:00                 4   
13690390  Kuala Lumpur 2023-08-28 01:48:37.945000+08:00              2216   
19056004  Kuala Lumpur 2023-08-28 17:35:44.997000+08:00                49   
20244015      Selangor 2023-08-28 22:16:24.330000+08:00                66   
21151475      Selangor 2023-08-28 12:07:07.003000+08:00                 3   

             content_type                         content  content_source  \
14622656              VOD     Lagu Didi & Friends S3 Ep31       astro.30d   
24953294              VOD             Upin & Ipin S5 Ep06       astro.30d   
12353742           LINEAR  (L) PL: Newcastle vs Liverpool        ASSP3 HD   
4993584            LINEAR        A Pillow Case of Mystery  TVB CLASSIC HD   
20949672              VOD     Lagu Didi & Friends S2 Ep08       astro.30d   
...                   ...                             ...             ...   
21510289  Play From Start                       Majalah 3             TV3   
13690390              VOD      Jangan Menangis Cinta Ep18       astro.30d   
19056004           LINEAR  Arsenal TV : Arsenal Vs Fulham     beIN SPORTS   
20244015              VOD         Where's Chicky? S2 Ep14        astro.7d   
21151475       Live Pause                     Berita 7:57      Awesome TV   

               date             time        timezone  \
14622656 2023-08-28  19:37:11.438000  Asia/Singapore   
24953294 2023-08-29  00:54:11.028000  Asia/Singapore   
12353742 2023-08-28  00:55:21.046000  Asia/Singapore   
4993584  2023-08-27  14:08:21.937000  Asia/Singapore   
20949672 2023-08-28  13:51:03.669000  Asia/Singapore   
...             ...              ...             ...   
21510289 2023-08-28  21:41:16.559000  Asia/Singapore   
13690390 2023-08-28  01:48:37.945000  Asia/Singapore   
19056004 2023-08-28  17:35:44.997000  Asia/Singapore   
20244015 2023-08-28  22:16:24.330000  Asia/Singapore   
21151475 2023-08-28  12:07:07.003000  Asia/Singapore   

                                 end_time     timestamp      day  
14622656 2023-08-28 19:39:10.438000+08:00  1.693223e+09   Monday  
24953294 2023-08-29 01:00:29.028000+08:00  1.693242e+09  Tuesday  
12353742 2023-08-28 01:27:50.046000+08:00  1.693155e+09   Monday  
4993584  2023-08-27 14:09:48.937000+08:00  1.693117e+09   Sunday  
20949672 2023-08-28 13:53:11.669000+08:00  1.693202e+09   Monday  
...                                   ...           ...      ...  
21510289 2023-08-28 21:41:20.559000+08:00  1.693230e+09   Monday  
13690390 2023-08-28 02:25:33.945000+08:00  1.693159e+09   Monday  
19056004 2023-08-28 17:36:33.997000+08:00  1.693215e+09   Monday  
20244015 2023-08-28 22:17:30.330000+08:00  1.693232e+09   Monday  
21151475 2023-08-28 12:07:10.003000+08:00  1.693196e+09   Monday  

[4378496 rows x 12 columns]

In [65]:
nz_duartion_df.shape

(4378496, 12)

In [66]:
(nz_duartion_df.shape[0] / main_df.shape[0]) * 100

16.832621692930797

In [67]:
sampled_df["state"].value_counts()

Kuala Lumpur       2752122
Selangor           1052497
Penang              301065
Johor               289287
Perak               161315
Sarawak              91277
Negeri Sembilan      87627
Sabah                83336
Melaka               47890
Kedah                37831
Kelantan             24785
Pahang               21128
Terengganu           12850
Putrajaya             2883
Perlis                 755
Labuan                 158
Name: state, dtype: int64

In [68]:
nz_duartion_df["state"].value_counts()

Kuala Lumpur       2439745
Selangor            927057
Penang              263188
Johor               254033
Perak               138457
Sarawak              81245
Negeri Sembilan      75902
Sabah                72932
Melaka               41072
Kedah                31033
Kelantan             21524
Pahang               18232
Terengganu           10718
Putrajaya             2525
Perlis                 706
Labuan                 127
Name: state, dtype: int64

## List of distinct content providers

In [69]:
provider_list = nz_duartion_df["content_source"].value_counts()

In [70]:
provider_list

astro.30d        1428113
astro.7d          570836
astro.com.my      322057
STAR VIJAY HD     175092
TV3               143212
                  ...   
CHUTTI TV              6
beIN SPORTS 4          3
astro.7d.ads           3
ARENA BOLA 3           2
Too FUN!               1
Name: content_source, Length: 183, dtype: int64

### top 20 content providers

In [72]:
provider_list.head(20) # seeing top 20 content providers

astro.30d         1428113
astro.7d           570836
astro.com.my       322057
STAR VIJAY HD      175092
TV3                143212
astro.2d           134957
ASSP4 HD           111529
ASSP3 HD            89780
ASSP HD             78008
SUN TV HD           59203
ZEE TAMIL HD        57529
AEC HD              56604
RIA HD              44760
beIN SPORTS         38814
CERIA HD            37328
Vellithirai HD      36054
VINMEEN HD          34765
TVBJ                32027
AXN HD              31854
ONE HD              29918
Name: content_source, dtype: int64

### bottom 20 content providers

In [73]:
provider_list.tail(20)

DW HD                 66
TA-DAA!               52
Boomerang             47
WAI FM                35
NAS FM                33
V FM                  29
ProjectHighCouncil    26
OPUS                  26
JAZZ                  24
ASYIK FM              23
astro.5d              20
CLASSIC ROCK          18
eGG Network           15
KENYALANG             14
BAYU                  14
CHUTTI TV              6
beIN SPORTS 4          3
astro.7d.ads           3
ARENA BOLA 3           2
Too FUN!               1
Name: content_source, dtype: int64

In [ ]:
! pip install pandas_profiling
from pandas_profiling import ProfileReport
sampled_df_profile = ProfileReport(sampled_df)
sampled_df_profile.to_file(output_file = '/content/sampled_df_profile.html')

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

In [ ]:
nz_duration_df_profile =  ProfileReport(nz_duration_df_profile)
nz_duration_df_profile.to_file(output_file = '/content/nz_duration_df_profile.html')